supabase/migrations/20250506183217_tag_permissions.sql

```diff
@@ -0,0 +1,31 @@
+CREATE TABLE IF NOT EXISTS "public"."tags" (
+    "id" uuid NOT NULL,
+    "created_at" timestamp with time zone DEFAULT "now"() NOT NULL,
+    "name" "text" NOT NULL,
+    "color" "text" NOT NULL,
+    "visible" boolean NOT NULL,
+    "user_id" uuid NOT NULL, 
```

Comment:
Can you change this to `profile_id`? The `user_id` is less present throughout the UI than `profile_id` (to make it easier to reuse components for viewing an "anonymous" user vs a "real" user), so there will be one less join required to retrieve the profiles for a user.

Please also add foreign key relationships to `profiles` and `classes`